tooling/metamodel-generator/src/main/java/org/hibernate/jpamodelgen/annotation/AnnotationMetaEntity.java

```diff
@@ -301,28 +322,63 @@ else if ( containsAnnotation( method, Constants.HQL, Constants.SQL, Constants.FI
 		initialized = true;
 	}
 
-	private void findSessionGetter(TypeElement type) {
-		if ( !containsAnnotation( type, Constants.ENTITY ) ) {
+	private @Nullable ExecutableElement findSessionGetter(TypeElement type) {
+		if ( !containsAnnotation( type, Constants.ENTITY ) || isPanacheType(type)) {
 			for ( ExecutableElement method : methodsIn( type.getEnclosedElements() ) ) {
 				if ( isSessionGetter( method ) ) {
-					dao = true;
-					sessionType = addDaoConstructor( method );
+					return method;
 				}
 			}
-			if ( !dao ) {
-				final TypeMirror superclass = type.getSuperclass();
-				if ( superclass.getKind() == TypeKind.DECLARED ) {
-					final DeclaredType declaredType = (DeclaredType) superclass;
-					findSessionGetter( (TypeElement) declaredType.asElement() );
+			final TypeMirror superclass = type.getSuperclass();
+			if ( superclass.getKind() == TypeKind.DECLARED ) {
+				final DeclaredType declaredType = (DeclaredType) superclass;
+				ExecutableElement ret = findSessionGetter( (TypeElement) declaredType.asElement() );
+				if(ret != null) {
+					return ret;
 				}
-				for ( TypeMirror superinterface : type.getInterfaces() ) {
-					if ( superinterface.getKind() == TypeKind.DECLARED ) {
-						final DeclaredType declaredType = (DeclaredType) superinterface;
-						findSessionGetter( (TypeElement) declaredType.asElement() );
+			}
+			for ( TypeMirror superinterface : type.getInterfaces() ) {
+				if ( superinterface.getKind() == TypeKind.DECLARED ) {
+					final DeclaredType declaredType = (DeclaredType) superinterface;
+					ExecutableElement ret = findSessionGetter( (TypeElement) declaredType.asElement() );
+					if(ret != null) {
+						return ret;
 					}
 				}
 			}
 		}
+		return null;
+	}
+
+	private boolean isPanacheType(TypeElement type) {
+		return isOrmPanacheType(type) || isReactivePanacheType(type);
+	}
+
+	private boolean isOrmPanacheType(TypeElement type) {
+		ProcessingEnvironment processingEnvironment = this.context.getProcessingEnvironment();
+		TypeElement panacheRepositorySuperType = processingEnvironment.getElementUtils().getTypeElement("io.quarkus.hibernate.orm.panache.PanacheRepositoryBase");
+		TypeElement panacheEntitySuperType = processingEnvironment.getElementUtils().getTypeElement("io.quarkus.hibernate.orm.panache.PanacheEntityBase");
+		if(panacheRepositorySuperType == null || panacheEntitySuperType == null) {
```

Comment:
Perhaps move these strings to the `Constants` class. It's not my personal preference at all, but I've been trying to be consistent with what the guy who came before me did.

Reply:
Fair.